sources/simulators.swift

```diff
@@ -1,12 +1,36 @@
 import Foundation
 
-enum SimulatorFetchError: String, Error {
-    case simctlFailed = "Running `simctl list` failed"
-    case failedToReadOutput = "Failed to read output from simctl"
-    case noBootedSimulators = "No simulators are currently booted"
+struct Simulator {
+    fileprivate enum State: String {
+        case shutdown = "Shutdown"
+        case booted = "Booted"
+    }
+
+    let name: String
+    let udid: String
+
+    fileprivate init?(dictionary: [String: String]) {
+        guard let state = dictionary["state"].map(State.init), let udid = dictionary["udid"],
+            let name = dictionary["name"], state == .booted else
```

Comment:
To me, this init is conflating json deserialization with a use case (`state == .booted`).

Reply:
moved it out of here